flexmeasures/data/models/reporting/__init__.py

```diff
@@ -0,0 +1,192 @@
+from typing import Optional, Union, Dict
+
+import pandas as pd
+from flask import current_app
+
+from flexmeasures.data.schemas.reporting import ReporterConfigSchema
+from flexmeasures.data.models.time_series import Sensor
+
+import timely_beliefs as tb
+
+
+class Reporter:
+    """Superclass for all FlexMeasures Reporters."""
+
+    __version__ = None
+    __author__ = None
+
+    sensor: Sensor = None
+
+    reporter_config: Optional[dict] = None
+    reporter_config_raw: Optional[dict] = None
+    schema = ReporterConfigSchema
+    data: Dict[str, Union[tb.BeliefsDataFrame, pd.DataFrame]] = None
+
+    def __init__(
+        self, sensor: Sensor, reporter_config_raw: Optional[dict] = None
+    ) -> None:
+        """
+        Initialize a new Reporter.
+
+        Attributes:
+        :param sensor: sensor where the output of the reporter will be saved to.
+        :param reporter_config_raw: unserialized configuration of the reporter.
+        """
+
+        self.sensor = sensor
+
+        if not reporter_config_raw:
+            reporter_config_raw = {}
+
+        self.reporter_config_raw = reporter_config_raw
+
+    def fetch_data(self):
+        """
+        Fetches the time_beliefs from the database
+        """
+
+        self.data = {}
+        for tb_query in self.tb_query_config:
+
+            # using start / end instead of event_starts_after/event_ends_before when not defined
+            event_starts_after = tb_query.pop("event_starts_after", self.start)
+            event_ends_before = tb_query.pop("event_ends_before", self.end)
+            resolution = tb_query.pop("resolution", self.input_resolution)
+
+            sensor = tb_query.pop("sensor", None)
+
+            bdf = sensor.search_beliefs(
+                event_starts_after=event_starts_after,
+                event_ends_before=event_ends_before,
+                resolution=resolution,
+                **tb_query,
+            )
+
+            # adding sources
+            for source in bdf.sources.unique():
+                self.data[f"source_{source.id}"] = source
+
+            # saving bdf
+            self.data[
+                f"sensor_{sensor.id}"
+            ] = bdf  # TODO: Add alias to reference dataframes easily. e.g: dict(sensor = 1, alias="power"),
+
+    def compute(self, *args, **kwargs) -> tb.BeliefsDataFrame:
```

Comment:
Please add `start` and `end` to the function signature explicitly.

You are adding these keys to the reporter_config (and also reporter_config_raw), which negates their static nature. I don't think we should do that. At least not to the raw config, which I think of as something of a (reasonably) static sensor attribute.

Reply:
Preferably, I'd also like to add `resolution` and `belief_time` explicitly, which, as discussed, matches functionality of the `Scheduler` class (although these parameters are for now still set in its `__init__` rather than in its `compute` method).